rust/rsc/src/common/database.rs

```diff
@@ -382,6 +383,21 @@ pub async fn create_many_output_dirs<T: ConnectionTrait>(
 // --------------------------------------------------
 // ----------             Blob             ----------
 // --------------------------------------------------
+pub async fn upsert_blob<T: ConnectionTrait>(
+    db: &T,
+    blob: blob::ActiveModel,
+) -> Result<Uuid, DbErr> {
+    let result = blob::Entity::insert(blob)
+        .on_conflict(
+            OnConflict::columns(vec![blob::Column::Key, blob::Column::StoreId])
+                .update_column(blob::Column::CreatedAt)
```

Comment:
We need to update the CreatedAt timestamp for eviction reasons?
It seems like it would be best to keep the the time of creation the same and just track the new use time

Reply:
We do yeah. Every blob has an unreferenced TTL where we guarantee that the blob won't get evicted during.

If we didn't then the following would occur.

Job A -- reference --> Blob 1
Job B uploads a  blob but its a dupe so id = 1 is returned
Job A gets evicted and now blob 1 is unreferenced and beyond its TTL so its eligible for eviction so its deleted
Job B gets uploaded referencing id = 1 which doesn't exist so the upload fails


Reseting the created_at also resets the TTL window

Reply:
Is it worth the effort to create a second entry to separate the eviction data from the "metrics data"? Or renaming "CreatedAt" to note that it is just the most recent time this blob was reference by a new job?

Reply:
Not opposed. Could be something like TTLWall or similar. Could be interesting to have both data points

Reply:
Ok I think its best to do that now so we bundle as many database updates together in a single release, but I'll leave it up to you whether you want to do that in this PR or a follow up.

Reply:
I'll do it in a follow up after merging the two queued up PRs